src/btree/row_modify.c

```diff
@@ -311,13 +311,13 @@ __wt_row_insert_alloc(WT_SESSION_IMPL *session, const WT_ITEM *key, u_int skipde
  */
 WT_UPDATE *
 __wt_update_obsolete_check(
-  WT_SESSION_IMPL *session, WT_PAGE *page, WT_UPDATE *upd, bool update_accounting)
+  WT_SESSION_IMPL *session, WT_PAGE *page, WT_UPDATE *upd, bool update_accounting, u_int *upd_count)
 {
     WT_TXN_GLOBAL *txn_global;
     WT_UPDATE *first, *next;
     size_t size;
     uint64_t oldest, stable;
-    u_int count, upd_seen, upd_unstable;
+    u_int upd_seen, upd_unstable;
 
```

Comment:
We should initialize upd_count to 0 here as a side effect of this function call.

Reply:
I have changed the design a bit, now doing the page-eviction check and schedule in `__wt_update_obsolete_check` function.